lib/rubocop/cop/rspec/capybara/specific_matcher.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Capybara
+        # Checks for there is a more specific matcher offered by Capybara.
+        #
+        # @example
+        #
+        #   # bad
+        #   expect(page).to have_selector('button')
+        #   expect(page).to have_none_of_selectors('button.cls')
+        #   expect(page).to have_css('button')
+        #   expect(page).to have_no_css('a.cls', exact_text: 'foo')
+        #   expect(page).to have_css('table.cls')
+        #   expect(page).to have_css('select')
+        #
+        #   # good
+        #   expect(page).to have_button
+        #   expect(page).to have_button(class: 'cls')
+        #   expect(page).to have_button
+        #   expect(page).to have_no_link('foo', class: 'cls')
+        #   expect(page).to have_table(class: 'cls')
+        #   expect(page).to have_select
+        #
+        class SpecificMatcher < Base
+          RESTRICT_ON_SEND = %i[have_selector have_none_of_selectors have_css
+                                have_no_css has_css? has_no_css?].freeze
+          SPECIFIC_MATCHER = {
+            button: 'button',
+            a: 'link',
+            table: 'table',
+            select: 'select'
+          }.freeze
+          MSG = 'Use `#%<good_matcher>s` instead of `#%<bad_matcher>s`.'
+
+          def on_send(node)
+            return unless node.first_argument.str_type?
+            return unless (matcher = specific_matcher(node))
+
+            add_offense(node, message: message(node, matcher))
+          end
+
+          private
+
+          def specific_matcher(node)
+            SPECIFIC_MATCHER.fetch(
+              node.first_argument.children.first.split(/\W/).first.to_sym, nil
```

Comment:
Cosmetic: with lengthy constructs like `node.first_argument.children.first`, it may make sense to consider using node pattern.